assignments/assignment02.rb

```diff
@@ -9,8 +9,23 @@
 
 # creates an english string from array
 
-def to_sentence(ary)
-  # your implementation here
+def to_sentence(ary, punctuate=false) # Optionally capitalizes the 1st word & appends a period (just for fun...)
+  new_string = ""
+  last_index = ary.length - 1
+  ary.each_with_index do |word, index|
+    if index == last_index
+      new_string << word.to_s
+    elsif index == ( last_index - 1 )
+      new_string << word.to_s << " and "
+    else
+      new_string << word.to_s << ", "
+    end
+  end
```

Comment:
This algorithm definitely works. However, you are doing multiple comparisons against index on each iteration. The alternative is to recognize that there are 2 special cases (the string contains 0 or 1 letters), test for those once, and then handle the normal case without doing all those tests.